### PR TITLE
dnn patches for slice, eltwise  and mvn layer

### PR DESCRIPTION
**Pull request by pengli**
```
allow_multiple_commits=1
```

### CODE REVIEW

File: modules/dnn/src/layers/eltwise_layer.cpp [hunk 1]
```diff
@@ -271,22 +272,32 @@ class EltwiseLayerImpl : public EltwiseLayer
...
-                    UMat mul0, mul1;
-                    multiply(coeffs[0], inputs[0], mul0);
-                    multiply(coeffs[1], inputs[1], mul1);
-                    add(mul0, mul1, outputs[0]);
-                    for (int i = 2; i < inputs.size(); ++i)
+                    int channels = outputs[0].size[0] * outputs[0].size[1];
+                    int plane_size = outputs[0].size[2] * outputs[0].size[3];
+                    size_t global[] = { (size_t)channels, (size_t)plane_size };
+                    int number = (plane_size % 8 == 0) ? 8 : ((plane_size % 4 == 0) ? 4 : 1);
+                    String buildopt = format("-DNUM=%d ", number);
```
Comment (alalek) [inline modules/dnn/src/layers/eltwise_layer.cpp]: Please avoid using of "extra" spaces. Apple OpenCL doesn't like that: #10710 .

**Comment by alalek** [comment]
@pengli There is ENet performance test slowdown (8% on i7-6700K). Could you take a look on this?

**Comment by pengli** [comment]
@alalek , ENet perf issue is fixed in the new eltwise patch

**Comment by alalek** [comment]
@pengli Please take a look on the failed accuracy test.

**Comment by pengli** [comment]
@alalek ,  Hi, the failed test is fixed.

**Pull request merged**